### PR TITLE
CEDS-1969 Send Extra auth headers to ILE for Internal Requests

### PR DESCRIPTION
**Pull request by ed0906**
This:
- Sends extra headers `X-Badge-Identifier` and `X-Submitter-Identifier` to the ILE API for internal submissions. Note we send fake data for these for now.
- Removes the `X-EORI-Identifier` as the ILE API doesn't use it
- Removes the `Request` suffix from `MovementRequest` and `ConsolidationRequest` as it added unnecessary clutter and polluted the whole stack with "Request" which is a Controller layer concern.
- Combines the `ILEMapper` and `WCOMapper`. They are both `ILEMapper`s really, one does `Consolidation`s, one does `Movement`s.

### CODE REVIEW

File: app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala [hunk 1]
```diff
@@ -21,10 +21,10 @@ import play.api.Logger
 import play.api.http.{ContentTypes, HeaderNames}
 import play.api.mvc.Codec
 import play.mvc.Http.Status
-import uk.gov.hmrc.exports.movements.config.AppConfig
+import uk.gov.hmrc.exports.movements.config.{AppConfig}
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala]: Unnecessary parenthesis in import

File: app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala [hunk 2]
```diff
@@ -34,33 +34,32 @@ import scala.xml.NodeSeq
...
-      HeaderNames.ACCEPT -> "application/vnd.hmrc.1.0+xml",
-      HeaderNames.CONTENT_TYPE -> ContentTypes.XML(Codec.utf_8),
-      CustomsHeaderNames.XClientIdName -> appConfig.clientIdInventory,
-      CustomsHeaderNames.XEoriIdentifierHeaderName -> eori
-    )
+  private def headers(identification: UserIdentification)(implicit hc: HeaderCarrier): Seq[(String, String)] = {
+    val authHeaders = identification.providerId match {
+      case Some(_) =>
+        Seq(
+          CustomsHeaderNames.BadgeIdentifier -> "ABC123",
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala]: Is 'ABC123' a placeholder for some later value?

File: app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala [hunk 3]
```diff
@@ -34,33 +34,32 @@ import scala.xml.NodeSeq
...
+          CustomsHeaderNames.SubmitterIdentifier -> "ABC123",
```
Comment (ed0906) [inline app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala]: As far as we are aware nothing validates these. We just need to send these headers for ILE APIs validation

File: app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala [hunk 4]
```diff
@@ -34,33 +34,32 @@ import scala.xml.NodeSeq
...
+          CustomsHeaderNames.XClientIdName -> appConfig.clientIdInventory
+        )
+      case _ => Seq(CustomsHeaderNames.XClientIdName -> appConfig.clientIdInventory)
+    }
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala]: I think that in situation where you don't need to have access to data there is no point of using pattern matching, simple `if` with` isDefined` can be better, but it's just personal opinion

File: test/integration/uk/gov/hmrc/exports/movements/connector/WiremockTestServer.scala [hunk 5]
```diff
@@ -31,14 +31,15 @@ trait WiremockTestServer extends WordSpec with BeforeAndAfterAll {
...
-    super.beforeAll()
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
     wireMockServer.start()
     WireMock.configureFor(wireHost, wirePort)
   }
 
-  override protected def afterAll(): Unit = {
+  override protected def afterEach(): Unit = {
+    wireMockServer.resetAll()
```
Comment (PatrykRudnicki) [inline test/integration/uk/gov/hmrc/exports/movements/connector/WiremockTestServer.scala]: Is it necessary to reset the `wireMockServer` if we're stopping it in next line?

File: app/uk/gov/hmrc/exports/movements/services/WCOMapper.scala [hunk 6]
```diff
@@ -62,4 +71,34 @@ class WCOMapper {
           transportNationality = Some(data.nationality)
       )
     )
+
+  def generateConsolidationXml(consolidation: Consolidation): Node =
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/movements/services/WCOMapper.scala]: I'm not sure that WCOMapper is a good place for it. It's not related with WCODec library

File: app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala (hunk 1 above)
Comment (ed0906) [reply app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala]: Suprised Scalafmt doesnt handle that

File: app/uk/gov/hmrc/exports/movements/services/WCOMapper.scala (hunk 6 above)
Comment (ed0906) [reply app/uk/gov/hmrc/exports/movements/services/WCOMapper.scala]: I'll rename it ILEMapper

File: app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala (hunk 2 above)
Comment (ed0906) [reply app/uk/gov/hmrc/exports/movements/connectors/CustomsInventoryLinkingExportsConnector.scala]: Nope, apparently we just need to populate the header, doesnt matter what with aslong as it meets their regex for that field. Conveniently `ABC123` matches both :)

**Pull request merged**